### ISSUE TITLE
[FlexibleHeader] Add an API for setting an explicit preferredStatusBarStyle

### ISSUE THREAD

**Issue by jverkoey** [material-components/material-components-ios#4747]
FlexibleHeader will attempt to infer the `preferredStatusBarStyle` using the MDFTextAccessibility library, which is rad. However, sometimes there's wiggle room and a client might want to explicitly use a white status bar instead of a black one, or vice-versa.

We should enable client teams to configure this accordingly using an explicit API.

The API should be optional and nil by default. If not provided, the flexible header will continue using its automatic behavior.

If a status bar style is provided however, then that status bar style will be used instead.

Because UIStatusBarStyle is a number type, we may have to model this as a Bool property, where if the Bool is enabled we use the explicitly-set status bar style, otherwise we use the inferring behavior
